tests/integration/test_interface.py

```diff
@@ -41,5 +43,13 @@ def test_generate_form(data_dir_name: str, noising_function: Callable):
     assert noised_data.equals(noised_data_same_seed)
     assert not noised_data.equals(noised_data_different_seed)
 
+    # Check each column's dtype
+    for col in noised_data.columns:
+        expected_dtype = [c.dtype for c in COLUMNS if c.name == col][0]
+        if expected_dtype == np.dtype(str):
+            # str dtype is 'object'
+            expected_dtype = np.dtype(object)
+        assert noised_data[col].dtype
```

Comment:
You're not checking that the actual matches the expected?

Reply:
Not sure I understand

Reply:
lol, oh, uh - yeah I guess I didn't quite finish this